proposals/feature-dependencies.md

```diff
@@ -0,0 +1,206 @@
+
+From an implementation point of view, `installsAfter` nodes may be added as a seperate set of directed edges, just as `dependsOn` nodes are added as directed edges (see **(B1)**).  Before round-based installation and sorting **(B2)**, an orchestrating tool should remove all `installsAfter` directed edges that do not correspond with a Feature in the `worklist` that is set to be installed.  In each round, a Feature can then be installed if all its requirements (both `dependsOn` and `installsAfter` dependencies) have been fulfilled in previous rounds.
+
+An implemention should fail the dependency resolution step if the evaluation of the `installsAfter` property results in an inconsistent state (eg: a circular dependency).
+
+### overrideFeatureInstallOrder
+
+The `overrideFeatureInstallOrder` property of `devcontainer.json` is an array of Feature IDs that are to be installed in descending priority order as soon as its dependencies outlined above are installed.  
+
+This evaluation is performed after **Round Stable Sort** and may result in a different installation order within a round.
+
+This property must not influence the dependency relationship as defined by the dependency graph (see **(B1)**) and shall only be evaulated at the round-based sorting step (see **(B2)**).  Put another way, this property cannot "pull forward" a Feature until all of its dependencies (both soft and hard) have been installed.  After a Feature's dependencies have been installed in other rounds, this property should "pull forward" each Feature as early as possible (given the order of identifiers in the array).
+
+Similar to `installsAfter`, this property's members may not provide options, nor are they able to be pinned to a specific version tag or digest.  This is unchanged from the current specification.
+
+If a Feature is indicated in `overrideFeatureInstallOrder` but not a member of the dependency graph (it is not queued to be installed), the orchestrating tool may fail the dependency resolution step.
+
+One implementation strategy involves assigning each node in the `worklist` a roundPriority integer, defaulting to 0.  If a Feature is indicated in `overrideFeatureInstallOrder`, the orchestrating tool should set the roundPriority of that Feature to  `length(overrideFeatureInstallOrder) - idx`, where `idx` is the zero-based index of the property in the array.  The orchestrating tool should then sort the `worklist` by roundPriority before committing each round's installation order, placing higher priority Features earlier in the `installationOrder`.
+
+## Additional Remarks
+
+### Feature authorship
+
+Features should be authored with the following considerations:
+
+- Features should be authored with the assumption that they will be installed in any order, so long as the dependencies are met at some point beforehand.
+- Since two Features with different options are considered different, a single Feature may be installed more than once.  Features should be idempotent.
```

Comment:
I am doubtful if the devcontainer/features can be installed multiple times. Curious, when do we update them? (before or after the CLI changes involving this spec)

Reply:
Without this property being included, the installation algorithm should behave as it did previously. Tactically, the Features in `devcontainers/features` will be updated and the major version incremented for them to take advantage of dependencies.  Other Features may choose as they please to adopt dependencies